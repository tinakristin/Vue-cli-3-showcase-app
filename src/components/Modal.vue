<template>
    <transition name="modal-fade">
        <div class="modal-backdrop">
            <div class="modal-content"
                 role="dialog"
                 aria-labelledby="modalTitle"
                 aria-describedby="modalDescription">
                <header class="modal-header">
                    <h2 id="modalTitle">
                        <slot name="headerText">
                            Dialog
                        </slot>
                        <button type="button"
                                class="btn-close pull-right"
                                aria-label="Lukk modal"
                                tabindex="0"
                                autofocus
                                @keyup.esc="close"
                                @click="close">x</button>
                    </h2>
                </header>
                <section class="modal-body" id="modalDescription">
                    <slot name="body"></slot>
                </section>
                <footer class="modal-footer">
                    <slot name="footer">
                        <button type="button"
                                class="btn ok-btn ok-btn-success"
                                role="button"
                                tabindex="0"
                                @click="confirm"
                                aria-label="Bekreft modal">
                            {{ $t('modal.confirm') }}
                        </button>
                        <button type="button"
                                class="btn ok-btn ok-btn-primary"
                                role="button"
                                tabindex="0"
                                @click="close"
                                aria-label="Lukk modal">
                            {{ $t('modal.decline') }}
                        </button>
                    </slot>
                </footer>
            </div>
        </div>
    </transition>
</template>

<script>
export default {
  name: 'Modal',
  methods: {
    close() {
      this.$emit('close');
    },
    confirm() {
      this.$emit('confirm');
    },
  },
};
</script>

<style scoped>
    .modal-backdrop {
        position: fixed;
        background-color: rgba(0, 0, 0, 0.6);
        display: flex;
        justify-content: center;
        align-items: center;
    }

    .modal-content {
        background: #FFFFFF;
        box-shadow: 2px 2px 20px 1px;
        overflow-x: auto;
        display: flex;
        flex-direction: column;
        width: 70%;
        max-width: max-content;
        height: min-content;
    }

    .modal-header,
    .modal-footer {
        padding: 15px;
        display: flex;
        width: 100%;
    }

    .modal-header {
        padding-bottom: 0;
        border-bottom: 1px solid #eeeeee;
    }

    #modalTitle {
        width: 100%;
    }

    .modal-footer {
        border-top: 1px solid #eeeeee;
        justify-content: flex-end;
    }

    .modal-body {
        position: relative;
        padding: 20px 10px;
    }

    .btn-close {
        align-self: flex-end;
        margin-left: auto;
        border: none;
        font-size: 20px;
        padding-left: 30px;
        cursor: pointer;
        font-weight: bold;
        background: transparent;
    }

    .modal-fade-enter,
    .modal-fade-leave-active {
        opacity: 0;
    }

    .modal-fade-enter-active,
    .modal-fade-leave-active {
        transition: opacity 0.5s ease;
    }
</style>
