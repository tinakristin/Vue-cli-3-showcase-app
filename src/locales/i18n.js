import Vue from 'vue';
import VueI18n from 'vue-i18n';
import translation from './nb-NO/translation.json';

Vue.use(VueI18n);

export default new VueI18n({
  locale: 'no', // set locale
  fallbackLocale: 'no',
  messages: {
    no: translation, // set locale messages
  },
});
