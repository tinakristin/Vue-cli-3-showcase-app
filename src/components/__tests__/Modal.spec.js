import { shallowMount } from '@vue/test-utils';
import i18n from '../../locales/i18n';
import Modal from '../Modal.vue';


describe('Modal', () => {
  it('matches the snapshot', () => {
    const wrapper = shallowMount(Modal, { i18n });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
