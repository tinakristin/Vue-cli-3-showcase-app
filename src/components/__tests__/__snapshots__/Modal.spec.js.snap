// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal matches the snapshot 1`] = `
<div class="modal-backdrop" name="modal-fade">
  <div role="dialog" aria-labelledby="modalTitle" aria-describedby="modalDescription" class="modal-content">
    <header class="modal-header">
      <h2 id="modalTitle">
        Dialog
        <button type="button" aria-label="Lukk modal" tabindex="0" autofocus="autofocus" class="btn-close pull-right">x</button></h2>
    </header>
    <section id="modalDescription" class="modal-body"></section>
    <footer class="modal-footer"><button type="button" role="button" tabindex="0" aria-label="Bekreft modal" class="btn ok-btn ok-btn-success">
        Ja
      </button> <button type="button" role="button" tabindex="0" aria-label="Lukk modal" class="btn ok-btn ok-btn-primary">
        Nei
      </button></footer>
  </div>
</div>
`;
